test/e2e_node/dra_test.go

```diff
@@ -231,22 +558,26 @@ func createTestObjects(ctx context.Context, clientSet kubernetes.Interface, node
 	createdPod, err := clientSet.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
 	framework.ExpectNoError(err)
 
-	ginkgo.DeferCleanup(clientSet.CoreV1().Pods(namespace).Delete, podName, metav1.DeleteOptions{})
+	if deferPodDeletion {
+		ginkgo.DeferCleanup(clientSet.CoreV1().Pods(namespace).Delete, podName, metav1.DeleteOptions{})
```

Comment:
You probably added `deferPodDeletion` because some test now needs to delete the pod and then this call failed.

You can remove the parameter, always do the DeferCleanup, and ignore "not found" errors:
```suggestion
		ginkgo.DeferCleanup(framework.IgnoreNotFound(clientSet.CoreV1().Pods(namespace).Delete), podName, metav1.DeleteOptions{})
```

https://github.com/kubernetes/kubernetes/blob/8c1983ffc0b6fe2293fc721cef8d961d79aafc53/test/e2e/framework/ginkgowrapper.go#L91-L96

Reply:
> You can remove the parameter, always do the DeferCleanup, and ignore "not found" errors:

Have I understood correctly that you're proposing to ignore pod deletion for all test cases, most of which don't delete a pod?

Reply:
Correct. This cleanup code only needs to ensure that the pod gets deleted. If that has been done already, then that's not a test failure.

Reply:
In most of the tests it still makes sense to check if pod is deleted, at least to my understanding. Pod is deleted in a couple of test cases only. Am I missing something?

Reply:
You always delete here. Then if the error is "NotFound", ignore the error because the desired state has already been reached.

Reply:
Ignoring the fact that pod doesn't exist when it should exist is worse than having a parameter in my opinion. WDYT?

Reply:
> You always delete here. Then if the error is "NotFound", ignore the error because the desired state has already been reached.

Not sure I follow. In most of test cases we shouldn't ignore "NotFound" error as the pod should exist. If it doesn't exist, then cleanup should fail.

Reply:
We don't need to be that strict. In other words, "unexpected pod deletion" is not something that we need to test for.

Reply:
I don't agree. We need to test for it explicitly, but we shouldn't regress tests quality just because we don't want to add one parameter.

Reply:
IMHO the `deferPodDeletion` should be the opposite: always delete to be on the safe side, but only treat "not found" in some tests.

But have it your way.